### ISSUE TITLE
Problems with header files

### ISSUE THREAD

**Issue by rsekman** [DeaDBeeF-Player/deadbeef#2844]
There are some issues with header files provided for plugins with public APIs (converter, artwork, gtkui).

- `converter.h`: It is installed in the same directory as `deadbeef.h`, but resides in `plugins/converter/` and consequently imports `../../deadbeef.h`. So the installed file cannot be included.
- `gtkui_api.h`: Uses several types without including their definitions. For example,
    https://github.com/DeaDBeeF-Player/deadbeef/blob/2cb64e6121bd191b97f9cdf2d8d20c48c61acfc8/plugins/gtkui/gtkui_api.h#L227 uses `gpointer` without including `glib.h`. It also uses some `ddb_` types   https://github.com/DeaDBeeF-Player/deadbeef/blob/2cb64e6121bd191b97f9cdf2d8d20c48c61acfc8/plugins/gtkui/gtkui_api.h#L206  https://github.com/DeaDBeeF-Player/deadbeef/blob/2cb64e6121bd191b97f9cdf2d8d20c48c61acfc8/plugins/gtkui/gtkui_api.h#L295-L297   without including `deadbeef.h`. Of course, anyone using `gtkui_api.h` would presumably include `deadbeef.h` themself, but it is inelegant to trust the third-party programmer to do things in the correct order.   Fixing this would create the same problem as for `converter.h` where the installed directory structure does not match the source tree.
-  `artwork.h`: also uses DDB types without including `deadbeef.h` https://github.com/DeaDBeeF-Player/deadbeef/blob/2cb64e6121bd191b97f9cdf2d8d20c48c61acfc8/plugins/artwork/artwork.h#L55    This does not cause issues building because `artwork.c` includes `deadbeef.h` *before* `converter.h`: https://github.com/DeaDBeeF-Player/deadbeef/blob/2cb64e6121bd191b97f9cdf2d8d20c48c61acfc8/plugins/artwork/artwork.c#L51-L54

To summarise, `converter.h` is unusable because installed directory tree does not match source tree, and the others require the third-party programmer to make sure dependencies are included in the correct order.

**Comment by Oleksiy-Yakovenko** [comment]
Thanks for pointing this out. The converter.h has a problem, it should not have the deadbeef.h include.
The other headers seem to be fine.

The expected usage is that whoever includes those files -- should also include the prerequisites.
It's done this way because the include paths may be different depending on the usage.
